javacord-core/src/main/java/org/javacord/core/entity/server/ServerImpl.java

```diff
@@ -821,6 +908,63 @@ public String getName() {
         return name;
     }
 
+    @Override
+    public Collection<ServerFeature> getFeatures() {
+        return serverFeatures;
+    }
+
+    @Override
+    public PremiumTier getPremiumTier() {
+        return premiumTier;
+    }
+
+    @Override
+    public int getPremiumSubscriptionCount() {
+        return premiumSubscriptionCount;
+    }
+
+    @Override
+    public Optional<Long> getRulesChannelId() {
+        return rulesChannelId != -1 ? Optional.of(rulesChannelId) : Optional.empty();
+    }
+
+    @Override
+    public Optional<String> getDescription() {
+        return Optional.ofNullable(description);
+    }
+
+    @Override
+    public Optional<Long> getPublicUpdatesChannelId() {
+        return publicUpdatesChannelId != -1 ? Optional.of(publicUpdatesChannelId) : Optional.empty();
+    }
+
+    @Override
+    public Optional<VanityUrlCode> getVanity() {
+        return Optional.ofNullable(vanityUrlCode);
+    }
+
+    @Override
+    public Optional<Icon> getDiscoverySplash() {
+        if (splash == null) {
+            return Optional.empty();
+        }
+        try {
+            return Optional.of(new IconImpl(
+                    getApi(),
+                    new URL("https://cdn.discordapp.com/discovery-splashes/" + getIdAsString() + "/" + discoverySplash + ".png")));
+        } catch (MalformedURLException e) {
+            logger.warn("Seems like the url of the icon is malformed! Please contact the developer!", e);
```

Comment:
See my comment in `VanityUrlCode`.
```suggestion
            throw new AssertionError("Unexpected malformed discovery splash url", e);
```